### ISSUE TITLE
Question:  How to add an existing SharePoint Site Column to a list

### ISSUE THREAD

**Issue by mikezimm** [pnp/pnpjs#1067]
### Category
- [ ] Enhancement
- [ ] Bug
- [ X ] Question
- [ ] Documentation gap/issue

### Version

Please specify what version of the library you are using: [    "@pnp/sp": "^2.0.0",     ]
Please specify what version(s) of SharePoint you are targeting: [    SPO     ]

I think this is just a quick question.

I'm trying to add the out of the box "Due Date" column (static name "TaskDueDate") to my list using this sample:   https://pnp.github.io/pnpjs/sp/fields/#add-a-site-field-to-a-list

However I think that example is expecting me to create a new site column first, not add an existing one like I want to do.

Based on the other samples in the docs, I tried getting the field, then adding it to the list fields, it's asking for more parameters like it is going to modify the field.

Is this the correct route to add an existing site column?

```typescript
const dueDate: IField = await sp.web.fields.getByInternalNameOrTitle('TaskDueDate');
//NOTE:  ensureResult.list does have the correct list at this point because this code is added after I successfully created some other columns.
const dueDateField = await ensureResult.list.fields.add('TaskDueDate', more stuff required....);
```

Thanks,

**Comment by ValerasNarbutas** [comment]
Try use **addDateTime** method instead of add, also check if duedate contains additional properties required for date type field:

**Comment by mikezimm** [comment]
@ValerasNarbutas ,
I tried that but what it does is create a brand new "non-sitecolumn" column.

To confirm, I went to into UI and tried "Add column from existing site columns", found it and added the real column I wanted.

Then I went in and tried to add the same column a second time but it was not available.  I did this just to make sure I could not add the site column twice which is what I expected of the real site column.

At that point, both columns existed on the list (my 'created' one and the one I wanted).

So then grabbed the Field Schema from the real column on the list... and will use that to create the actual field I want...

Here's my code... it works, but I was just hoping to not have to use the schema route because well... more complex code.

But thanks for the reply.  I'll close the issue.

```typescript
fieldSchema = '<Field ID="{cd21b4c2-6841-4f9e-a23a-738a65f99889}" Name="TaskDueDate" Group="Core Task and Issue Columns" Type="DateTime" DisplayName="Due Date" SourceID="http://schemas.microsoft.com/sharepoint/v3/fields" StaticName="TaskDueDate" Format="DateOnly" DelayActivateTemplateBinding="GROUP,SPSPERS,SITEPAGEPUBLISHING" AllowDeletion="TRUE" ColName="datetime2" RowOrdinal="0" />';

const dueDateField: IFieldAddResult = await ensureResult.list.fields.createFieldAsXml(fieldSchema);
```
